src/prob/pf.jl

```diff
@@ -198,3 +198,803 @@ function build_mc_pf(pm::AbstractUBFModels)
         constraint_mc_transformer_power(pm, i)
     end
 end
+
+
+## Native power flow solver
+"""
+internal data required used solving an ac unbalanced power flow
+the primary use of this data structure is to prevent re-allocation of memory
+between successive power flow solves
+* `data_math` -- a PowerModelsDistribution mathematical data dictionary
+* `ntype` -- node type: @enum NodeType
+* `cc_ns_func_pairs` -- a mapping of nodes to compensation current evaluation functions 
+* `indexed_nodes` -- all nodes in the network that are not grounded
+* `node_to_idx` -- a mapping of all nodes (bus, terminal) to a unique value
+* `fnode_to_idx` -- a mapping of nodes with fixed voltage (bus, terminal) to a unique value
+* `vnode_to_idx` -- a mapping of nodes with variable voltage (bus, terminal) to a unique value
+* `fixed_nodes` -- all indexed nodes with a known voltage reference
+* `Uf` -- vector of known voltages
+* `Yf` -- admittance matrix partition: Y[node_other_idx, node_fixed_idx]
+* `Yv` -- admittance matrix partition: Y[node_other_idx, node_other_idx]
+The postfix `_idx` indicates the admittance matrix indexing convention.
+"""
+mutable struct PowerFlowData
+    data_math::Dict
+    ntype::Dict
+    cc_ns_func_pairs::Vector{Tuple}
+    indexed_nodes::Vector
+    node_to_idx::Dict
+    fnode_to_idx::Dict
+    vnode_to_idx::Dict
+    fixed_nodes::Vector
+    Uf::Vector
+    Yf::Matrix
+    Yv::Matrix
+end
+
+"""
+    PowerFlowData(
+      data_math::Dict,
+      v_start::Dict
+    )
+
+Constructor for PowerFlowData struct that requires mathematical model and v_start.
+
+v_start assigns the initialisation voltages to appropriate bus terminals.
+"""
+function PowerFlowData(data_math::Dict{String,<:Any}, v_start::Dict{<:Any,<:Any}, explicit_neutral::Bool)
```

Comment:
missing property for explicit_neutral in the docstring

Reply:
added: explicit_neutral indicates that the neutral conductor is explicitly modelled.